### PR TITLE
802.1q support

### PR DESCRIPTION
**Pull request by adrianchiris**

### CODE REVIEW

File: pkg/tc/generator.go [hunk 1]
```diff
@@ -111,22 +127,34 @@ func (s *SimpleTCGenerator) genFilters(ipCidrs []*net.IPNet, ports []policyrules
 	case hasIPs:
 		for _, ipCidr := range ipCidrs {
 			var proto tctypes.FilterProtocol
-			var actualPrio uint16
+			var ipProtoPrio uint16
+			var vlanPotoPrio = prio + 2
```
Comment (ykulazhenkov) [inline pkg/tc/generator.go:131]: vlanPotoPrio > vlanProtoPrio

File: pkg/tc/generator.go [hunk 2]
```diff
@@ -98,9 +102,21 @@ func (s *SimpleTCGenerator) genDropFilters(rule policyrules.Rule) []tctypes.Filt
...
+		tctypes.NewGenericActionBuiler().WithDrop().Build())
+}
+
 // genFilters generates (flower) Filters based on provided ipCidrs, ports on the given prio with the given action
 // the filters generated are: matching on {ipCidrs} [X {Ports}] With priority `prio`, and action `action`
-// if no IPs and Ports provided, returned filters will match all ipv4, ipv6 traffic
+// if no IPs and Ports provided, returned filters will match all ipv4, ipv6, 802.1q traffic with provided action
+//
+//nolint:funlen
 func (s *SimpleTCGenerator) genFilters(ipCidrs []*net.IPNet, ports []policyrules.Port, prio uint16,
```
Comment (ykulazhenkov) [inline pkg/tc/generator.go:120]: I think it is better to refactor this function. We already have https://github.com/k8snetworkplumbingwg/multi-networkpolicy-tc/issues/42, so not a blocker

File: pkg/tc/generator.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (adrianchiris) [reply pkg/tc/generator.go:131]: was planning to address this as part of #42

File: pkg/tc/generator.go [hunk 3]
```diff
@@ -148,27 +185,68 @@ func (s *SimpleTCGenerator) genFilters(ipCidrs []*net.IPNet, ports []policyrules
...
 			for idx, proto := range allProtocols {
 				actualPrio := prio + uint16(idx)
+
+				if proto == tctypes.FilterProtocol8021Q {
+					// for vlan protocol we need to match on both ipv4 and ipv6 vlan eth type
+					filters = append(filters,
+						tctypes.NewFlowerFilterBuilder().
+							WithProtocol(proto).
+							WithPriority(actualPrio).
+							WithMatchKeyVlanEthType("ip").
```
[the remark this answers is not in the document]
Comment (adrianchiris) [reply pkg/tc/generator.go:195]: was planning to address this part of #41

**Review by ykulazhenkov — APPROVED** [review]
refactoring required, will be addressed by #41, #42

**Pull request merged**